nucypher/blockchain/eth/signers.py

```diff
@@ -191,3 +216,93 @@ def unlock_account(self, account: str, password: str, duration: int = None) -> b
     @validate_checksum_address
     def lock_account(self, account: str):
         return True
+
+
+class KeyStoreSigner(Signer):
+
+    IS_SIGNER_LOCAL = True
+
+    def __init__(self, uri: str):
+        super().__init__()
+
+        decoded_uri = urlparse(uri)
+        if decoded_uri.scheme not in ('key', 'keystore') or decoded_uri.netloc:
+            raise cls.InvalidSignerURI(uri)
+
+        with open(decoded_uri.path) as file_obj:
+            key = json.load(file_obj)
+
+        address = key['address']
+        if not Web3.isAddress(address):
+            raise ValueError(f"'{decoded_uri.path}' does not contain a valid address")
+
+        self.key = key
+        self._account = Web3.toChecksumAddress(address)
```

Comment:
The single underscore already indicates it's private, mangling the name with a second underscore shouldn't be necessary (and stil doesn't make it const either), but if that extra emphasis is desired, I have absolutely no objection.

The other members should also be private, particularly self.key which is an implementation detail.

web3.Web3 doesn't export to_checksum_address() just toChecksumAddress() - as per published api is here:
https://web3py.readthedocs.io/en/stable/overview.html#Web3.toChecksumAddress

Of course looking at the web3 code web3/main imports to_checksum_address from eth_utils, which Web3 then exposes as toChecksumAddress(), so you can call the same underlying function any of these ways:

* eth_utils.to_checsum_address
* web3.main.to_checksum_address
* web3.Web3.toChecksumAddress

Which of the two underscore_names is preferable?

Reply:
> The single underscore already indicates it's private, mangling the name with a second underscore shouldn't be necessary (and stil doesn't make it const either), but if that extra emphasis is desired, I have absolutely no objection.

I'd prefer double underscore here to prevent setting the attribute explicitly; I also noticed this attribute is exposed as a `@property` which suggests the "read-only" nature of the namespace mangling.

> Which of the two underscore_names is preferable?

For consistency with the rest of the codebase, use `eth_utils.to_checsum_address`.

Reply:
Done - member names double underscored, camel case functions replaced with eth_utils.*.